cdap-app-fabric/src/main/java/io/cdap/cdap/internal/app/services/AppFabricServer.java

```diff
@@ -171,11 +172,19 @@ protected void startUp() throws Exception {
     }
 
     cancelHttpService = startHttpService(httpServiceBuilder.build());
-    long count = TransactionRunners.run(transactionRunner,
-                                        (TxCallable<Long>) context ->
-                                          AppMetadataStore.create(context).getApplicationCount());
+    Long[] applicationAndNamespaceCount = TransactionRunners.run(transactionRunner,
+                                                                 context ->
+                                                                   new Long[] {
+                                                                     AppMetadataStore.create(context)
+                                                                       .getApplicationCount(),
+                                                                     new DefaultNamespaceStore(transactionRunner)
```

Comment:
DefaultNamespaceStore.getNamespaceCount() already runs in a transaction. Why are we wrapping it in another transaction?

Reply:
Done. Thanks for catching; I think it was from previous logic when there was no transaction in the class I was using.